browser/ui/views/frame/brave_browser_frame_view_win.cc

```diff
@@ -105,3 +106,25 @@ int BraveBrowserFrameViewWin::NonClientHitTest(const gfx::Point& point) {
 
   return result;
 }
+
+int BraveBrowserFrameViewWin::FrameTopBorderThickness(bool restored) const {
+  if (tabs::features::HorizontalTabsUpdateEnabled()) {
+    if (!(frame()->IsFullscreen() || IsMaximized()) || restored) {
+      if (browser_view()->GetTabStripVisible()) {
+        return 2;
+      }
+    }
+  }
+  return BrowserFrameViewWin::FrameTopBorderThickness(restored);
+}
+
+int BraveBrowserFrameViewWin::TopAreaHeight(bool restored) const {
+  if (tabs::features::HorizontalTabsUpdateEnabled()) {
+    if (!frame()->IsFullscreen() || restored) {
+      if (!IsWebUITabStrip()) {
```

Comment:
The logic here is a bit tough, but I tried to replicate the behavior of the patches exactly.